include/arch/arm/aarch32/syscall.h

```diff
@@ -163,19 +165,37 @@ static inline uintptr_t arch_syscall_invoke0(uintptr_t call_id)
 	return ret;
 }
 
+static inline bool aarch32_is_unprivileged_state(void)
+{
+	uint32_t value;
+
+#if defined(CONFIG_CPU_CORTEX_R)
+	__asm__ volatile("mrs %0, CPSR\n\t" : "=r"(value));
+
+	/*
+	 * For Cortex-R, the mode bits (lower 5 bits) will be 0x10 for user mode.
+	 */
+	return ((value & 0x1f) == 0x10) ? true : false;
+#else
+	/* return mode information */
+	__asm__ volatile("mrs %0, CONTROL\n\t" : "=r"(value));
```

Comment:
Can you use the Cortex-M CMSIS function, __get_CONTROL() here?

Reply:
fixed